activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -351,6 +351,19 @@ def supports_pgcrypto_uuid?
         postgresql_version >= 90400
       end
 
+      def supports_optimizer_hints?
+        unless defined?(@has_pg_hint_plan)
+          unless extension_enabled?("pg_hint_plan")
+            enable_extension("pg_hint_plan")
```

Comment:
Why it needs to enable and them disable?

Reply:
If `extension_enabled?("pg_hint_plan")` is true, the extension is already enabled, otherwise the extension is not enabled, and we don't know whether or not the extension is available.

The `enable_extension` is to determine the extension is available or not (if not available, raise the `StatementInvalid` error), then the `disable_extension` is for keeping enabled extensions as before.

Reply:
You can check via `SELECT * FROM pg_available_extensions;` or `SELECT * FROM pg_extension;`.

Reply:
Good to know! I've confirmed it works as expected.